### PR TITLE
VSHN-managed LBs

### PR DESCRIPTION
**Pull request by simu**
This PR replaces the MVP LB implementation with Puppet-managed LBs using `profile_openshift4_gateway`.

Note that this PR currently contains hard-coded configurations which only make sense for VSHN-managed setups.

### CODE REVIEW

File: lb.tf [hunk 1]
```diff
@@ -39,9 +80,92 @@ data "exoscale_compute_template" "ubuntu2004" {
...
+      # hieradata.
+      SERVER_ZONE = replace(var.region, "-", "_")
+      # Cluster id is used as encdata stage
+      CLUSTER_ID = var.cluster_id
+    }
+  }
+}
+
+resource "gitfile_checkout" "appuio_hieradata" {
+  repo = "https://${var.hieradata_repo_user}@git.vshn.net/appuio/appuio_hieradata.git"
```
Comment (ccremer) [inline lb.tf:113]: Is it intentional that this terraform module would then only be able to install LBs for the APPUiO tenant?
For other tenants/customers, we wouldn't be able to install loadbalancers?

**Review by ccremer — APPROVED** [review]
LGTM, though I did not do a full in-depth review.

File: lb.tf (hunk 1 above)
Comment (simu) [reply lb.tf:113]: As far as I understand, the OCP4 managed LBs will always be provisioned in the APPUiO tenant, even when the cluster belongs to another tenant.

Comment (tobru) [reply lb.tf:113]: Yes, that's the idea, as the LBs are a black-box for tenant of the cluster.

File: files/commit-hieradata.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+fi
+
+if [ "${push}" -eq 1 ]; then
+  push_args=
+  if [ "${amend}" -eq 1 ]; then
+    push_args="--force"
+  fi
+  git push ${push_args} origin "${branch}"
+  # Set local branch to track upstream branch at origin
+  git branch -u "origin/${branch}"
```
Comment (corvus-ch) [inline files/commit-hieradata.sh:52]: Is there a reason not to put the `-u` option on `git push`?

File: lb.tf [hunk 3]
```diff
@@ -54,106 +178,65 @@ resource "exoscale_compute" "lb" {
...
+  # Privnet CIDR IPs starting from .21
+  # (IPs .1,.2,.3 will be assigned by Puppet profile_openshift4_gateway)
+  ip_address = cidrhost(var.privnet_cidr, 21 + count.index)
+}
+
+resource "exoscale_domain_record" "lb" {
+  count  = var.lb_count
+  domain = exoscale_domain.cluster.id
+  name   = random_id.lb[count.index].hex
+  # TBD
```
Comment (corvus-ch) [inline lb.tf:238]: What is missing here?

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Please update the README. I would like to see a general description of what this module does. Also it needs an explanation of the context it requires (control.vshn.net, puppet, git.vshn.net, …).

File: lb.tf (hunk 3 above)
Comment (simu) [reply lb.tf:238]: I initially wasn't sure if a TTL of 10 minutes is reasonable for the LB DNS records, but I think it's good enough for now.

File: files/commit-hieradata.sh (hunk 2 above)
Comment (simu) [reply files/commit-hieradata.sh:52]: I just forgot `push -u` is a thing, fixed in the latest version

**Comment by simu** [comment]
> Please update the README. I would like to see a general description of what this module does. Also it needs an explanation of the context it requires (control.vshn.net, puppet, git.vshn.net, …).

I've added a short README.

**Review by corvus-ch — APPROVED** [review]
Thanks, now looking good.

**Pull request merged**